proposals/0000-android-12-issues.md

```diff
@@ -0,0 +1,364 @@
+
+## Impact on existing code
+
+### Foreground Services
+
+The RouterService currently sends an intent out for each apps BroadcastReceiver which notifies the apps to start the `SdlService`. If the app targets Android 12 and tries to start the `SdlService` from the background like this the app will crash. We will need to add a PendingIntent as an extra to this intent to be used by the `SdlReceiver`
+These changes will require app developers to change how they implement the `SdlReceiver.onReceive` method to check the incoming intent for a pendingIntent and send that pendingIntent with the appropriate class name.
+This implementation will also require the `SdlService` to have the `android:exported` attribute set to `true` in the `AndroidManifest.xml`.
+
+### Bluetooth Runtime Permissions
+
+The bluetooth logic in the library will now require these 2 new runtime permissions. With these permissions granted there is not any issue with the bluetooth logic but we now need to account for the use cases where a user has denied permissions. Specifically without these permissions the `BroadcastReceiver` will never receive Intents with the `ACL_CONNECTED` action and the `RouterService` will not be able to start the bluetoothTransport.
+
+### AndroidManifest Exported Flag
+
+These attributes were already defined where required. 
+
+### PendingIntent Mutable Flag
+
+The RouterService will crash if the the app targets android 12 and does not have this flag defined on the PendingIntent used for RouterService notifications.
+
+### Service Notification Delays
+
+These notifications may be delayed by Android by up to 10 seconds
+
+## Alternatives considered
+
+### Foreground Services
+
+Alternatives for the Foreground Service restrictions are limited. We either need to start the `SdlService` from a foreground context or the conditions need to meet one of the exceptiions listed by google. These conditions include:
+
+* Starting the Service from an Activity.
+* Starting the Service from a user interaction with a notification
+* Requesting the user ignores battery optimizations for each SDL Application
+
+These options would either recquire and Activity be launched for each SDL app, the user to interact with a notification for each SDL app, or for the user to choose battery optimization options for each app which then creates a situation where the user dictates if the given app will have its `SdlService` start when the RouterService conencts.
+
+### Bluetooth Runtime Permissions
+
+These permissions are required to be able to receive intents withe the `ACL_CONNECTED` action in the BroadcastReceiver and to start the bluetoothTransport in the RouterService. Without these permissions we cannot know when the device connects to a device over Bluetooth nor will the RouterService be able to connect over Bluetooth.
+
+### AndroidManifest Exported Flag
```

Comment:
If there are no possible alternatives considered, I suggest removing that section